File: java-sdk/src/main/java/com/globalpayments/api/gateways/IPaymentGateway.java
package com.globalpayments.api.gateways;

public interface IPaymentGateway {
    Object processAuthorization(Object builder) throws Exception;
    Object manageTransaction(Object builder) throws Exception;
    Object processReport(Object builder) throws Exception;
    Object serializeRequest(Object builder) throws Exception;
    boolean supportsOpenBanking();
}

File: node-sdk/src/Gateways/Interfaces/IPaymentGateway.ts
import { Transaction } from "../../Entities/Transaction";

export interface IPaymentGateway {
  supportsHostedPayments: boolean;
  supportsOpenBanking: boolean;

  processAuthorization(builder: any): Promise<Transaction>;
  manageTransaction(builder: any): Promise<Transaction>;
  serializeRequest(builder: any): string;
}
